contrib/buildrpm/buildrpmLibfabric.sh

```diff
@@ -0,0 +1,368 @@
+    cp -fp "$tarball" "$rpmbuilddir/SOURCES/"
+  fi
+else
+  rm_cmd="rm -rfv \"$rpmbuilddir\""
+  mkdir_cmd="mkdir -pv \"$rpmbuilddir/SOURCES/\""
+  cp_cmd="cp -fpv \"$tarball\" \"$rpmbuilddir/SOURCES/\""
+  if [[ -z "$noop" ]]; then
+    echo "$0: --> Cleanup old rpmbuild directory"
+    eval "$rm_cmd"
+    echo "$0: --> Create rpmbuild SOURCES directory"
+    eval "$mkdir_cmd"
+    echo "$0: --> Copy tarball to rpmbuild directory"
+    eval "$cp_cmd"
+  else
+    echo "$0: --> Cleanup old rpmbuild directory with command:"
+    echo "$0: --> $rm_cmd"
+    echo "$0: --> Create rpmbuild SOURCES directory using command:"
+    echo "$0: --> $mkdir_cmd"
+    echo "$0: --> Copy tarball to rpmbuild directory with command:"
+    echo "$0: --> $cp_cmd"
+  fi
+fi
+
+###########
+# Build RPM
+###########
+export cmd=""
+export build_opt=""
+
+if [[ -z "$verbose" ]]; then
+  build_opt="--quiet"
+else
+  build_opt="-v"
+fi
+cmd="rpmbuild $build_opt -bb $specfile $rpmbuild_options \
+  --define 'configopts $configure_options' \
+  --define '_topdir $rpmbuilddir' \
+  --define '_sourcedir $rpmbuilddir/SOURCES' \
+  --define '_rpmdir $rpmbuilddir/RPMS' \
+  --define '_specdir $rpmbuilddir/SPECS' \
+  --define '_tmppath $rpmbuilddir/tmp'"
+
+if [[ -n "$verbose" ]]; then
+  echo "$0: --> Build command used:"
+  echo "$0: --> $cmd"
+  echo "$0: --> RPM build will start"
+fi
+
+if [[ -z "$noop" ]]; then
+  eval "$cmd"
+
+  export status="$?"
+  if [[ "$status" != 0 ]]; then
+    echo "$0: FATAL: *** FAILURE BUILDING RPM! ERROR CODE: $status" 1>&2
+    echo "$0: FATAL: Aborting" 1>&2
+    exit 6
+  fi
+fi
+if [[ -n "$verbose" ]]; then
+  echo "$0: --> Done building the RPM"
+fi
+
+###################
+# Create modulefile
+###################
+if [[ -n "$create_modulefile" ]]; then
+if [[ -z "$noop" ]]; then
+echo "#%Module -*- tcl -*-
+#
+# Modulefile for Libfabric
+#
+
+proc ModulesHelp { } {
+   puts stderr \"\tThis module adds Libfabric $version to the environment.\"
+}
+
+module-whatis   \"This module adds Libfabric $version to the environment\"
+
+prepend-path PATH $prefix/bin
+prepend-path LD_LIBRARY_PATH $prefix/lib64
+prepend-path MANPATH $prefix/share/doc
+" > $rpmbuilddir/$version.tcl
+fi
+if [[ -n "$verbose" ]]; then
+	echo "$0: --> Done creating modulefile"
+fi
+fi
+
+##############
+# Print result
+##############
+if [[ -n "$verbose" ]]; then
+cat <<EOF
+
+------------------------------------------------------------------------------
+====                FINISHED BUILDING Libfabric RPM                       ====
+------------------------------------------------------------------------------
+A copy of the tarball is located in: $rpmbuilddir/SOURCES/
+The completed RPMs are located in:   $rpmbuilddir/RPMS/
+EOF
+
+if [[ -n "$create_modulefile" ]]; then
+  echo "The modulefile is located in:        $rpmbuilddir/$version.tcl"
+fi
+
+cat <<EOF
+------------------------------------------------------------------------------
+
+EOF
+fi
```

Comment:
Fair enough.  But I think you should still indent everything else, even if you can't indent the EOF (so that it properly ends the heredoc notation).